src/utils/spo.spec.ts

```diff
@@ -262,6 +262,62 @@ describe('utils/spo', () => {
     });
   });
 
+  it('retrieves tenant app catalog url', async () => {
+    auth.service.spoUrl = 'https://contoso.sharepoint.com';
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === 'https://contoso.sharepoint.com/_api/SP_TenantSettings_Current') {
+        return Promise.resolve({ CorporateCatalogUrl: 'https://contoso.sharepoint.com/sites/appcatalog' });
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+
+    const tenantAppCatalogUrl = await spo.getTenantAppCatalogUrl(logger, false);
+    assert.deepEqual(tenantAppCatalogUrl, 'https://contoso.sharepoint.com/sites/appcatalog');
+  });
+
+  it('retrieves tenant app catalog url when not set', async () => {
+    auth.service.spoUrl = 'https://contoso.sharepoint.com';
+    sinon.stub(request, 'get').callsFake((opts) => {
+      if (opts.url === 'https://contoso.sharepoint.com/_api/SP_TenantSettings_Current') {
+        return Promise.resolve({ CorporateCatalogUrl: null });
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    const tenantAppCatalogUrl = await spo.getTenantAppCatalogUrl(logger, false);
+    assert.deepEqual(tenantAppCatalogUrl, null);
+  });
+
+  it('handles error when retrieving SPO URL failed while retrieving application customizer', (done) => {
+    const errorMessage = 'An error has occurred';
+    auth.service.spoUrl = undefined;
+
+    sinon.stub(request, 'get').callsFake(async (opts) => {
+      if ((opts.url as string).indexOf('/_api/SP_TenantSettings_Current') > -1) {
+        return Promise.reject(errorMessage);
+      }
+
+      return Promise.reject('Invalid request');
+    });
+
+    spo
+      .getTenantAppCatalogUrl(logger, false)
+      .then(() => {
+        done('Expected error');
+      }, (err: string) => {
+        try {
+          assert.strictEqual(err, 'Invalid request');
```

Comment:
Shouldn't the expected error be the error message that we're returning on line 300?

Reply:
Hi @waldekmastykarz, 
When retrieving SPO URL fails, it will throw an exception and `getTenantAppCatalogUrl` will not get called.

Reply:
I see, and you're right. I suggest that we make the test more robust and define a specific error for the SPO URL request. Right now, 'Invalid request' will be returned for any other API call than tenant settings, which could obfuscate possible errors in our code.

Reply:
Hi @waldekmastykarz 
I did not get any specific error message here to show. But will `Invalid hostname for this tenancy` do?

Reply:
Let's use `Couldn't retrieve SharePoint URL`. After all, we had an error retrieving the URL so we can't tell if the URL is invalid or not.